mongodbatlas/resource_mongodbatlas_database_user.go

```diff
@@ -53,6 +53,11 @@ func resourceMongoDBAtlasDatabaseUser() *schema.Resource {
 				Optional: true,
 				Default:  "NONE",
 			},
+			"aws_iam_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "NONE",
+			},
```

Comment:
```suggestion
			"aws_iam_type": {
				Type:     schema.TypeString,
				Optional: true,
				Default:  "NONE",
				ValidateFunc: validation.StringInSlice([]string{"NONE", "USER", "ROLE"}, false),
			},
```
if only these attribute must to be set in this attribute, you need to consider put this validation to ensure that the user write at least one of them.

Reply:
Thanks for the suggestion, paco, I added it and for x509 too